### PR TITLE
Add support for handling Notion API character limits and preserving Markdown formatting

### PR DESCRIPTION
**Pull request by igorhlr**
# Enhanced Markdown Support for Notion Integration

## Overview

This PR implements support for bypassing Notion API's 2000-character block limit, allowing the creation and update of pages with content of any size while preserving Markdown formatting.

## Problem Being Solved

The Notion API has three main limitations that affect JotDown:

1. **Character limit of 2000 per text block**: Prevents creating pages with extensive content
2. **Lack of Markdown formatting**: Pages are created as simple text blocks without preserving Markdown formatting
3. **Limit of 100 blocks per request**: Prevents creation of complex pages in a single request

## Solution

This PR implements a solution that addresses these limitations by:

1. **Splitting content into smaller chunks**: Content is divided into parts that respect the 2000-character limit, preferably at natural break points like headers
2. **Converting Markdown to Notion blocks**: Each chunk is converted into appropriate Notion blocks, preserving the Markdown formatting (headers, lists, code blocks, etc.)
3. **Sending blocks in batches**: Formatted blocks are sent in batches of up to 100 to comply with the Notion API limitations

### New Module: `formatter.rs`

Added a new module to handle content splitting and formatting:
- `split_content`: Divides text into parts that respect the character limit
- `simple_split`: Helper method for dividing text without headers
- `format_for_notion`: Converts Markdown text into Notion blocks
- `get_valid_notion_language`: Maps language names to values accepted by Notion

### Modifications to `notion.rs`

Added new methods to work with multiple blocks:
- `create_page_with_blocks`: Creates a page with multiple formatted blocks
- `append_blocks`: Adds blocks to an existing page
- `update_page_with_blocks`: Updates a page with new formatted blocks

### Modifications to `jot.rs`

Updated the methods that interact with Notion to use the new functionality:
- `create_new_page`: Now splits, formats, and sends content as blocks
- `update_page`: Now splits, formats, and sends content as blocks

### Added Tests

Implemented tests for the main functionality:
- Tests for `split_content` with different input types
- Tests for `format_for_notion` to verify correct formatting

## Benefits

1. **Support for extensive content**: Users can create Notion pages with content of any length
2. **Preservation of Markdown formatting**: Formatting is maintained in the conversion to Notion blocks
3. **Better user experience**: Eliminates errors related to character limits
4. **Compatibility with the Notion API**: Respects API limitations while providing greater flexibility

## Inspired By

This implementation was inspired by similar functionality in the NotionAiAssistant project, adapted to JotDown's context and needs.

## Current Limitations & Future Improvements

The current implementation supports:
- Headers (H1, H2, H3)
- Bulleted and numbered lists
- Code blocks with language detection
- Preservation of entire code blocks (without splitting)

Future improvements could include:
- Support for more Markdown elements (tables, images, quotes, etc.)
- Caching for Notion references
- Performance optimizations for very large content

### CODE REVIEW

File: src/formatter.rs [hunk 1]
```diff
@@ -0,0 +1,373 @@
...
+        let chunks = split_content(text, 100);
+        assert_eq!(chunks.len(), 1);
+        assert_eq!(chunks[0], text);
+    }
+    
+    #[test]
+    fn test_split_content_large_text_no_headers() {
+        let text = "A".repeat(3000); // Large text without headers
+        let chunks = split_content(&text, 1000);
+        assert!(chunks.len() > 1);
```
Comment (Harry-027) [inline src/formatter.rs:366]: It looks like this test case is failing — possibly due to an edge case we overlooked when using a simple text split. Could you please take a closer look?

**Review by Harry-027 — COMMENTED** [review]
Minor nitpick – I think we can clean up unused methods like `create_page` in jot.rs, and both `update_page` and `create_page` in `notion.rs`. Other than that, I’m happy to merge this PR. Great contribution in working around the Notion API limits — keep it up!

**Review by Harry-027 — COMMENTED** [review]
Minor nitpick – I think we can clean up unused methods like `create_page` in jot.rs, and both `update_page` and `create_page` in `notion.rs`. Other than that, I’m happy to merge this PR. Great contribution in working around the Notion API limits — keep it up!

File: src/formatter.rs (hunk 1 above)
Comment (igorhlr) [reply src/formatter.rs:335]: Thanks for the detailed review and feedback!

Cleanup of unused methods:
I'll remove the create_page methods in jot.rs and update_page/create_page in notion.rs, since they've been replaced by the new implementations.

Test test_split_content_large_text_no_headers:
I agree that there's an edge case here. The simple_split method is probably not ensuring accurate splits when there are no headers. I'll adjust the logic to split the text into smaller chunks even without natural markers, ensuring that the character limit is respected.

Thank you very much for your willingness to approve the PR and for the acknowledgement! I'll send the corrections as soon as possible.

If you have any other questions, just let me know! 🙌

**Comment by igorhlr** [comment]
Hi @Harry-027,

Thank you for your review! I've implemented the requested changes:

1. Fixed the failing test case `test_split_content_large_text_no_headers` by improving the `simple_split` function to handle edge cases with plain text that has no natural splitting points
2. Removed the unused methods:
   - `create_page` from `jot.rs`
   - `update_page` and `create_page` from `notion.rs`

The test now passes successfully. Please let me know if there's anything else I should address!

**Comment by Harry-027** [comment]
Hi @igorhlr 
Thanks for the quick updates! The changes look good, and it's great to see the test passing now. Appreciate you cleaning up the unused methods as well.

Merging this in now — great work! 🥇 🚀
